### ISSUE TITLE
Kafka - Unexpected auth request 3 expected handshaed

### ISSUE THREAD

**Issue by anggutie-dev** [redpanda-data/helm-charts#307]
Hello guys,

Today I deployed the latest helm chart for redpanda, I configured SASL 256, everyting is working as expected but getting this message:

kafka - connection_context.cc:396 - Error processing request: kafka::kafka_authentication_exception (requests.cc:184 - Unexpected auth request 3 expected handshake)

I'm not really sure if I need anything else besides the SASL configuration (username & password).

**Comment by joejulian** [comment]
Just as a first step, I'll drop this here. I'm not trying to just say rtfm but in case you hadn't seen it, there have been a lot of updates to the docs recently: https://docs.redpanda.com/docs/manage/kubernetes/security/sasl-kubernetes/

Can you share your values overrides file or your helm command if you're using `--set`? (obviously don't share your user data)

**Comment by anggutie-dev** [comment]
Thanks for quick response @joejulian, here is my current values.yaml file

```
redpanda:

  # Change the default name
  fullnameOverride: "redpanda"

  # Default kuberentes cluster domain
  clusterDomain: cluster.local

  statefulset:
    # Number of Redpanda brokers (recommend setting this to the number of nodes in the cluster)
    replicas: 3

    # Add tolerations for redpanda nodepool
    tolerations:
      - key: "redpanda"
        operator: "Equal"
        value: "true"
        effect: "NoSchedule"

    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchExpressions:
          - key: redpanda
            operator: In
            values:
            - "true"
  
  #
  #
  # Authentication
  auth:
    #
    # SASL configuration
    sasl:
      enabled: true
      # user list
      # TODO create user at startup
      users:
        # Admin User
        - name: admin
          password: 
          mechanism: SCRAM-SHA-256

  #
  #
  # Enable TLS
  tls:
    enable: true
    certs:
      default:
        issuerRef:
          name: letsencrypt
          kind: ClusterIssuer

  #
  #
  # External access configuration
  external:
    domain: yieldbalance.com
    enabled: false
    addresses:
      - rpd0
      - rpd1
      - rpd2

  #
  #
  # Listeners
  listeners:
    kafka:
      external:
        defaults:
          port: 9092
          advertisedPorts:
          - 9092

  resources:
    # Both Redpanda and Kubernetes have multiple ways to allocate resources.
    # There are also several associated parameters that impact how these resources are used by
    # Kubernetes, the Redpanda app, and the subsystem Redpanda is built on (Seastar).
    # This section attempts to simplify allocating resources by providing a single location
    # where resources are defined.
    # Helm sets these resource values within the following templates:
    # - statefulset.yaml
    # - configmap.yaml
    #
    # The default values below are what should work for a development environment.
    # Production-level values and other considerations are provided in comments
    # if those values are different from the default.
    #
    cpu:
      # Redpanda makes use of a thread per core model described here:
      # https://redpanda.com/blog/tpc-buffers
      # For this reason, Redpanda should only be given full cores (cores parameter below).
      #
      # NOTE: You can increase cores, but decreasing cores is not currently supported:
      # https://github.com/redpanda-data/redpanda/issues/350
      #
      # Equivalent to: --smp, resources.requests.cpu, and resources.limits.cpu
      # For production: 4 or greater
      cores: 1
      #
      # Overprovisioned means Redpanda won't assume it has all of the provisioned CPU.
      # This should be true unless the container has CPU affinity (eg. min and max above are equal).
      # Equivalent to: --idle-poll-time-us 0 --thread-affinity 0 --poll-aio 0
      # overprovisioned: false
    #
    memory:
      # Enables memory locking.
      # For production: true
      enable_memory_locking: true
      #
      # It is recommended to have at least 2Gi of memory per core for the Redpanda binary.
      # This memory is taken from the total memory given to each container.
      # We allocate 80% of the container's memory to Redpanda, leaving the rest for
      # the Seastar subsystem (reserveMemory) and other container processes.
      # So at least 2.5Gi per core is recommended in order to ensure Redpanda has a full 2Gi.
      #
      # These values affect --memory and --reserve-memory flags passed to Redpanda and the memory
      # requests/limits in the StatefulSet.
      # Valid suffixes: k M G Ki Mi Gi
      # Only support a single decimal (eg. 2.5Gi rather than 2.55Gi)
      #
      container:
        # Minimum memory count for each Redpanda broker
        # If omitted, the min value will equal the max value (requested resources defaults to limits)
        # Equivalent to: resources.requests.memory
        # For production: 10Gi or greater
        min: 2.5Gi
        #
        # Minimum memory count for each Redpanda broker
        # Equivalent to: resources.limits.memory
        # For production: 10Gi or greater
        max: 2.5Gi

  #
  # Persistence
  storage:
    # Absolute path on host to store Redpanda's data.
    # If not specified, then `emptyDir` will be used instead.
    # If specified, but `persistentVolume.enabled` is `true`, then has no effect.
    hostPath: ""
    # If `enabled` is `true` then a PersistentVolumeClaim will be created and
    # used to store Redpanda's data, otherwise `hostPath` is used.
    persistentVolume:
      enabled: true
      size: 100Gi
      # If defined, then `storageClassName: <storageClass>`.
      # If set to "-", then `storageClassName: ""`, which disables dynamic
      # provisioning.
      # If undefined or empty (default), then no `storageClassName` spec is set,
      # so the default provisioner will be chosen (gp2 on AWS, standard on
      # GKE, AWS & OpenStack).
      storageClass: ""
      # Additional labels to apply to the created PersistentVolumeClaims.
      labels: {}
      # Additional annotations to apply to the created PersistentVolumeClaims.
      annotations: {}
  
  config:
    cluster:
      auto_create_topics_enabled: true
```

I'm basically creating the admin user, once I have this user ready I start creating users for my service with the proper ACL for each one.


Thanks again!

**Comment by alejandroEsc** [comment]
This may have been resolved by #496 could you verify please?

**Comment by joejulian** [comment]
Since we expect this is resolved and there's no further feedback, I'm closing this.
